### PR TITLE
codegen: nullable types

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: src/graphqlHelpers.ts [hunk 1]
```diff
@@ -891,7 +902,7 @@ export function typeScriptDefinitionObjectForFragment(
             namedFragments: [],
             inlineFragments: [],
             selections: {
-              displayedName: {
+              [displayedName]: {
```
Comment (anmonteiro) [inline src/graphqlHelpers.ts:905]: @sgrove this is a drive-by fix, I think you didn't intend to start with `displayedName` literally.

File: src/graphqlHelpers.ts [hunk 2]
```diff
@@ -524,6 +528,7 @@ export function typeScriptDefinitionObjectForOperation(
 
         let gqlType = field.type;
         let namedType = getNamedType(gqlType);
+        const isNullable = isNullableType(gqlType);
```
Comment (anmonteiro) [inline src/graphqlHelpers.ts:531]: we propagate nullable at the field level

File: src/graphqlHelpers.ts [hunk 3]
```diff
@@ -731,7 +742,7 @@ const printObject = (obj: OutObject): string => {
 `
         : "";
 
-      return `${description}${fieldSelection.name}: ${value};`;
+      return `${description}${fieldSelection.name}${fieldSelection.isNullable ? "?" : ""}: ${value};`;
```
Comment (anmonteiro) [inline src/graphqlHelpers.ts:745]: and use `isNullable` on the field to decide whether to emit `?` or not in the object definition.

**Review by sgrove — APPROVED** [review]
